operations/document-compression-updater/updater.py

```diff
@@ -0,0 +1,271 @@
+            print("result {}".format(result))
+            maxObjectIdToTouch = id["_id"]
+        lastScannedObjectId = 0
+        numDocumentsUpdated = 0
+           
+        numExistingDocuments = col.estimated_document_count()
+        
+            
+        first_entry = {
+            "collection_name": appConfig['collectionName'],
+            "lastScannedObjectId" : lastScannedObjectId,
+            "ts": datetime.datetime.now(tz=datetime.timezone.utc),
+            "maxObjectIdToTouch" : maxObjectIdToTouch,
+            "numExistingDocuments" : numExistingDocuments,
+            "numDocumentsUpdated" : numDocumentsUpdated
+            # scan fields in future, for now we use  _id
+            }   
+            
+        tracker_col.insert_one(first_entry)
+
+        printLog("create first entry in tracker db  for collection {}".format(first_entry),appConfig)
+            
+    client.close()
+
+    returnData = {}
+    returnData["numExistingDocuments"] = numExistingDocuments
+    returnData["maxObjectIdToTouch"] = maxObjectIdToTouch
+    returnData["lastScannedObjectId"] = lastScannedObjectId
+    returnData["numDocumentsUpdated"] = numDocumentsUpdated
+    
+    print(returnData)
+
+    return returnData
+
+
+def task_worker(threadNum,perfQ,appConfig):
+
+
+    maxObjectIdToTouch = appConfig['maxObjectIdToTouch']
+    lastScannedObjectId = appConfig['lastScannedObjectId']
+    numInsertProcesses = appConfig['numInsertProcesses']
+
+    
+    numExistingDocuments = appConfig["numExistingDocuments"]
+    maxObjectIdToTouch = appConfig["maxObjectIdToTouch"]
+    lastScannedObjectId = appConfig["lastScannedObjectId"]
+    numDocumentsUpdated = appConfig["numDocumentsUpdated"]
+
+    client = pymongo.MongoClient(appConfig['uri'])
+    myDatabaseName = appConfig['databaseName']
+    db = client[myDatabaseName]
+    myCollectionName = appConfig['collectionName']
+    col = db[myCollectionName]
+    
+    
+    tracker_db=client['tracker_db']
+    trackerCollectionName = myCollectionName+'_tracker_col'
+    tracker_col=tracker_db[trackerCollectionName]
+    
+    
+    allDone = False
+
+    tempLastScannedObjectId = lastScannedObjectId
+    
+    print(appConfig)
+
+    while not allDone:
+
+        #start and go through all the docs using _id 
+        
+        if lastScannedObjectId != 0 :
+       
+            batch =  col.find({"_id" : { "$gt" : lastScannedObjectId   }},{ "_id" :1}).sort({"_id" :1}).limit(appConfig['batchSize'])
+        else :
+            
+            batch =  col.find({},{ "_id" :1}).sort({ "_id" :1}).limit(appConfig['batchSize'])
+            
+            
+        batch_count = 0
+        
+
+        for id in batch : 
+
+            if id["_id"]<=maxObjectIdToTouch:
+                
+                # print("found id {} lesser than maxObjectIdToTouch {}.".format(str(id["_id"]),str(maxObjectIdToTouch)))
+                col.update_one({ "_id" : id["_id"] } , { "$set": { appConfig['updateField']: 1 } } )
```

Comment:
This needs to be changed to use a batch operation, not individual updates. There will be far too many round trips.